File: solid-kata/java/lsp-solution/src/main/java/com/codurance/lsp/PetrolCar.java
package com.codurance.lsp;

public class PetrolCar extends Vehicle implements PetrolPowered {
  private static final int FUEL_TANK_FULL = 100;
  private int fuelTankLevel = 0;

  @Override
  public void fillUpWithFuel() {
    this.fuelTankLevel = FUEL_TANK_FULL;
  }

  public int fuelTankLevel() {
    return fuelTankLevel;
  }
}

File: solid-kata/c#/LSP/PetrolCar.cs
namespace LSP
{
    public class PetrolCar : Vehicle
    {
        private const int FuelTankFull = 100;
        public int FuelTankLevel { get; private set; }

        public override void FillUpWithFuel() => FuelTankLevel = FuelTankFull;

        public override void ChargeBattery() => throw new InvalidOperationException("A petrol car cannot be recharged");
    }
}